code/game/objects/items/weapons/ipc_scanner.dm

```diff
@@ -0,0 +1,33 @@
+/obj/item/ipc_tag_scanner
+	name = "IPC tag scanner"
+	desc = "A hand-held IPC tag scanner, that, when used to analyze the info of an IPC, will output its tag status and information."
+	icon = 'icons/obj/ipc_utilities.dmi'
+	icon_state = "ipc_tag_scanner"
+	item_state = "ipc_tag_scanner"
+	contained_sprite = TRUE
+	flags = CONDUCT
+	slot_flags = SLOT_BELT
+	throwforce = 3
+	w_class = ITEMSIZE_SMALL
+	throw_speed = 5
+	throw_range = 10
+	origin_tech = list(TECH_MAGNET = 2, TECH_BIO = 1, TECH_ENGINEERING = 2)
+	matter = list(DEFAULT_WALL_MATERIAL = 500, MATERIAL_GLASS = 200)
+
+/obj/item/ipc_tag_scanner/attack(mob/living/M, mob/living/user)
+	add_fingerprint(user)
+	if(!isipc(M))
```

Comment:
You should move that check after the do_after to avoid easy meta-gaming by just rapidly clicking random people with the analyzer and seeing if it actually gives a "can only be used on IPCs" message.